### ISSUE TITLE
Bug: depending on how the deferredQuery is built, sometimes Neotypes puts extra spaces

### ISSUE THREAD

**Issue by jGonzalezFernandez** [neotypes/neotypes#526]
Hey guys, I just tried the library and I noticed that sometimes Neotypes puts extra spaces when you write a query. For example:

`  
  "CypherStringInterpolator" should {

    "put too many spaces??" in {

      val RETURN: DeferredQueryBuilder = c"RETURN"
      val n: String                    = "n"

      assert(c"$RETURN#$n".query.query == " RETURN  n ") // But it should be: "RETURN n"
    }

  }
`

Notice that there's an extra space before the RETURN, and two spaces between the RETURN and the n. This doesn't really matter because Neo4j ignores the extra spaces, but it would certainly look ugly in the logs.

I didn't investigate why this happens, but a quick win could be: do a systematic trim of the final result of interpolations, and also replace the extra spaces with just one. Something like:

`
query.trim.replaceAll(" +", " ")
`

**Comment by BalmungSan** [comment]
Yeah is a bit annoying, I noticed it when looking at some of the **cypher** tests but to be honest avoiding the space is not that trivial, and the `replace` feels wasteful for something that doesn't affect the end result of the query.

About the argument of the log, while it is compelling I wonder if big / complex queries won't have more logging issues anyways?

**Comment by tjarvstrand** [comment]
Yes, I was also a bit frustrated by this when writing the `DeferredQuery`-interpolation but realized pretty quickly that it's rather complicated to do something about.

There's now also the literal interpolator (`#${...}`) to consider, which may contain any whitespace, and people may also want to keep the indentation of their queries
